src/scenes/Authentication/Register/RegisterForm.tsx

```diff
@@ -0,0 +1,128 @@
+import { makeStyles, Typography } from '@material-ui/core';
+import { Mutator } from 'final-form';
+import React from 'react';
+import { Form, FormProps } from 'react-final-form';
+import {
+  blurOnSubmit,
+  EmailField,
+  focusLastActiveFieldOnSubmitError,
+  PasswordField,
+  SubmitButton,
+  SubmitError,
+  TextField,
+} from '../../../components/form';
+import { CordIcon } from '../../../components/Icons';
+import { Link } from '../../../components/Routing';
+
+const useStyles = makeStyles(({ spacing }) => ({
+  header: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  icon: {
+    fontSize: 64,
+    margin: 'auto',
+    marginBottom: spacing(4),
+  },
+  formError: {
+    margin: spacing(2, 0),
+  },
+  submit: {
+    marginTop: spacing(1),
+  },
+  loginLink: {
+    display: 'block',
+    marginTop: spacing(1),
+  },
+  footer: {
+    maxWidth: 500,
+    margin: spacing(3, 0, 2),
+    textAlign: 'center',
+  },
+  footerSpacer: {
+    visibility: 'hidden',
+    '@media (max-height: 700px)': {
+      display: 'none',
+    },
+  },
+  verse: {
+    fontStyle: 'italic',
+  },
+  reference: {
+    marginTop: spacing(1),
+  },
+}));
+
+interface RegisterInput {
+  firstName: string;
+  lastName: string;
+  email: string;
+  password: string;
+}
+
+export type RegisterFormProps = Pick<
+  FormProps<RegisterInput>,
+  'onSubmit' | 'initialValues'
+> & { className?: string };
+
+export const RegisterForm = ({ className, ...props }: RegisterFormProps) => {
+  const classes = useStyles();
+  return (
+    <>
+      <div className={className}>
+        <div className={classes.header}>
+          <CordIcon className={classes.icon} />
+          <Typography variant="h4" gutterBottom={true}>
+            CORD FIELD
+          </Typography>
+          <Typography color="textSecondary">
+            Accelerating Bible Translation
+          </Typography>
+        </div>
+        <Form
+          {...props}
+          decorators={decorators}
+          mutators={{ clearSubmitErrors }}
+        >
+          {({ handleSubmit }) => (
+            <form onSubmit={handleSubmit}>
+              <SubmitError className={classes.formError} />
+              <TextField
+                name="firstName"
+                label="First Name"
+                placeholder="Enter First Name"
+                required
+                autoFocus
+              />
+              <TextField
+                name="lastName"
+                label="Last Name"
+                placeholder="Enter Last Name"
+                required
+              />
+              <EmailField />
+              <PasswordField />
```

Comment:
I think we probably need a second confirm password field. I've done the same thing in 7748eb91dc62aa40e25512e60bc3bf97052eac81. It's a lot of code to duplicate, but it's only once so I guess it's fine.

Reply:
10-4